### ISSUE TITLE
`gfm_auto_identifiers` does not remove . in headers leading to non-working link in Github 

### ISSUE THREAD

**Issue by cderv** [jgm/pandoc#8709]
````
> pandoc -t gfm --toc -s
# A header

# With ending dot.

# With a.dot in the middle
^Z
- [A header](#a-header)
- [With ending dot.](#with-ending-dot.)
- [With a.dot in the middle](#with-a.dot-in-the-middle)

# A header

# With ending dot.

# With a.dot in the middle
````

Dots are not removed whereas `gfm_auto_identifiers` should remove them. From https://pandoc.org/MANUAL.html#extension-gfm_auto_identifiers

> Extension: gfm_auto_identifiers
> Changes the algorithm used by auto_identifiers to conform to GitHub’s method. Spaces are converted to dashes (-), uppercase characters to lowercase characters, and **punctuation characters other than - and _ are removed**. Emojis are replaced by their names.

This will produce non working link on Github. See this [gist](https://gist.github.com/cderv/84a7f00b5de2f7c09c1005b4c75751d1)

Tested with
``` 
> pandoc --version
pandoc.exe 3.1.1
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: C:\Users\chris\AppData\Roaming\pandoc
Copyright (C) 2006-2023 John MacFarlane. Web:  https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.
```

**Comment by jgm** [comment]
Here's what I'm seeing:
```
% pandoc -f gfm
# With ending dot.

# With a.dot in the middle
<h1 id="with-ending-dot">With ending dot.</h1>
<h1 id="with-adot-in-the-middle">With a.dot in the middle</h1>
```
Note that the dot is removed.  Are you seeing something different?

I think you're expecting pandoc to rewrite the identifiers you use in these links:

```
- [A header](#a-header)
- [With ending dot.](#with-ending-dot.)
- [With a.dot in the middle](#with-a.dot-in-the-middle)
```

That is not part of the documented behavior of this extension, and I wouldn't support rewriting explicitly given identifiers.

**Comment by cderv** [comment]
I am trying to convert a document to GFM 
This is Pandoc's markdown
```markdown
# With ending dot.

# With a.dot in the middle
```
and I want to convert of  GFM to create for example a `README.md` for Github.  So I am using
```
pandoc -f markdown -t gfm
```

I did not specify here some id using `# title {#id}` syntax. So I would expect the id created for the TOC to be valid for GFM. 
Using `pandoc -f markdown -t gfm --toc -s`. This is what I shared above. The links created for the TOC are not valid links to be used in GFM because the section links will have `.` removed. 

Hope it is clearer. 

From your example, I am understanding that  `+gfm_auto_idenfiers` is kind of a reader only option.  So I do this it outputs as expected 

````powershell
> pandoc -f markdown+gfm_auto_identifiers -t gfm --toc -s
# With ending dot.

# With a.dot in the middle
^Z
- [With ending dot.](#with-ending-dot)
- [With a.dot in the middle](#with-adot-in-the-middle)

# With ending dot.

# With a.dot in the middle
````

Is this the way it is expected to work ? This remind me of some issues in the pass with this. Sorry for asking again if this is really the expected way. 

If so, I find `--toc` not be useful for `gfm` output as it won't be valid TOC for Github, and one need to know that the extension needs to be used on the input format. Probably when `--toc` is handled, the ids have already been parsed by the reader with the gfm options. Maybe this should be added into the documentation to be clearer.

**Comment by jgm** [comment]
Sorry, I missed the fact that you're converting *to* gfm, even though it was in your original message.

**Comment by jgm** [comment]
You are right, the `gfm_auto_identifiers` option needs to be used with the reader, since the reader is adding the attributes to the AST.

I see why this is a bit awkward, but I'm not sure I see a better solution.  The writer has no way of knowing whether a given identifier was added as an auto identifier or given explicitly.
